winetricks: remove wine dependency

vs49688 opened the pull request:
###### Motivation for this change

Wintricks shouldn't be tied to a particular Wine installation, it should use the one in $PATH. Especially after https://github.com/NixOS/nixpkgs/pull/144085.
Also massively reduces the closure size.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
  - [x] winetricks
  - [ ] grapejuice - I can't test this
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

###### Review
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>grapejuice</li>
    <li>winetricks</li>
  </ul>
</details>

kira-bruneau commented on pkgs/misc/emulators/wine/winetricks.nix:
```diff
@@ -11,7 +11,7 @@ stdenv.mkDerivation rec {
   # coreutils is for sha1sum
   pathAdd = lib.concatMapStringsSep ":" (x: x + "/bin")
     (lib.filter (x: x != null)
```
I added the filter so I could remove the wine dependency in protontricks, it probably isn't necessary anymore with this change.

vs49688 replied:
Good point, have changed to `lib.makeBinPath`.

kira-bruneau reviewed (approved):
Thanks @vs49688, this makes sense!

vs49688 committed:
winetricks: remove wine dependency

It shouldn't be tied to a particular Wine installation, it should use
the one in $PATH. Also massively reduces the closure size.

rnhmjoj reviewed (approved):
This will make winetricks less reproducible, but being reproducible makes no sense in this case. In fact, it's even problematic if winetricks wine is 32bit only and your wine is a WoW version.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/should-we-support-encoding-runtime-dependencies-without-requiring-a-specific-implementation/21031/6
